cmd/genji/shell/command.go

```diff
@@ -136,3 +142,193 @@ func displaySuggestions(in string) error {
 	fmt.Println()
 	return nil
 }
+
+// dumpTable displays the content of the given table as SQL statements.
+func dumpTable(tx *genji.Tx, t *database.Table) error {
+	var buf bytes.Buffer
+
+	c := fmt.Sprintf("CREATE TABLE %s", t.Name())
+	buf.WriteString(c)
+
+	ti, err := t.Info()
+	if err != nil {
+		return err
+	}
+
+	fcs := ti.FieldConstraints
+	hasField := false
+	for i, fc := range fcs {
+		// Fields constraints should be displaying between parenthesis.
+		if !hasField {
+			buf.WriteString("(\n")
+			hasField = true
+		}
+
+		// Don't put the last comma.
+		if i > 0 {
+			buf.WriteString(",\n")
+		}
+
+		buf.WriteString("  " + fcs[i].Path.String() + " ")
+		buf.WriteString(strings.ToUpper(fcs[i].Type.String()))
+		if fc.IsPrimaryKey {
+			buf.WriteString(" PRIMARY KEY")
+		}
+
+		if fc.IsNotNull {
+			buf.WriteString(" NOT NULL")
+		}
+	}
+
+	// Fields constraints close parenthesis.
+	if hasField {
+		buf.WriteString("\n);\n")
+	} else {
+		buf.WriteString(";\n")
+	}
+
+	// Print CREATE TABLE statement.
+	_, err = fmt.Fprintf(os.Stdout, buf.String())
+	if err != nil {
+		return err
+	}
+	buf.Reset()
+
+	// Indexes statement.
+	indexes, err := t.Indexes()
+	if err != nil {
+		return err
+	}
+
+	for _, index := range indexes {
+		u := ""
+		if index.Opts.Unique {
+			u = " UNIQUE"
+		}
+
+		info := fmt.Sprintf("CREATE%s INDEX %s ON %s (%s);\n", u, index.Opts.IndexName, index.Opts.TableName,
```

Comment:
There is no reason for using the buffer here, you can do this in one action with `fmt.Printf`

Reply:
Right 👍